### DISCUSSION TITLE
DateRangePicker - Calendars do not updated properly.

### DISCUSSION THREAD

**Discussion by alex-airlaw** [rsuite/rsuite#2609] [Q&A]
Hello Team,

I added custom range as below and pass it into DateRangePicker component. 
It allowed for selection in the ranges menu but when I select it the month is not changed in the calendars in accordance with date range selected.
```
  const ranges: RangeType[] = [
    {
      label: 'This quarter',
      value: [
        moment().quarter(moment().quarter()).startOf('quarter').toDate(),
        moment().quarter(moment().quarter()).endOf('quarter').toDate(),
      ],
    },
}
```

BTW is there any way to access to calendars exactly?

Thanks.

**Comment by simonguo** [comment]
Could you please provide a complete reproduction example via `codesandbox`?

**Reply by alex-airlaw** [reply]
Hi simonguo, 
Thanks for reply.
I have created a simple example, you can try it here [SandBox](https://codesandbox.io/embed/wizardly-buck-0kgit0?fontsize=14&hidenavigation=1&theme=dark)
As you can see on screenshot below, the 'Last year' range is selected and it reflected above of calendars in the header but the calendars dates did not changed.
![image](https://user-images.githubusercontent.com/107915335/180949904-4c7ec5a8-3ba6-4cfe-8b83-0744c23d8d4a.png)

Thanks, 
Alex.

**Reply by simonguo** [reply]
`onOk` is triggered after clicking the OK button. You should use `onChange` instead of `onOk`.  In addition, you need to add a `closeOverlay:true` to the Range, the interactive experience will be better.

Fixed example: https://codesandbox.io/s/jovial-pond-444wzy

**Reply by alex-airlaw** [reply]
Thanks for possible way to solve the issue, but in your case after I selected the range the date-picked closing. I use `onOk`  because I need to change the dates only after user click "OK" button in other case initial date should be leaved.

**Reply by alex-airlaw** [reply]
Is it possible somehow to rerender calendars when `onChange` occurred, based on the new value?
